docs/getting-started/sdk/password-manager/web/interoperability.md

```diff
@@ -0,0 +1,269 @@
+
+```typescript
+/**
+ * @returns {number}
+ */
+export function do_something(): number;
+
+/**
+ * @returns {string}
+ */
+export function say_hello(): string;
+```
+
+### Structs
+
+If you want to return more complex types, you can annotate a Rust struct with the `wasm_bindgen`
+attribute. Make sure that any fields you want to access from JavaScript are public:
+
+```rust
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub struct User {
+    pub age: u32,
+}
+```
+
+Which generates the following TypeScript definitions:
+
+```typescript
+export class User {
+  free(): void;
+  age: number;
+}
+```
+
+For structs containing `String` or other types that are not `Copy`, use the `getter_with_clone`
+attribute to ensure JavaScript can access the value without directly referencing it in WebAssembly
+memory:
+
+```rust
+#[wasm_bindgen(getter_with_clone)]
+pub struct User {
+    pub name: String,
+    pub age: u32,
+}
+```
+
+Which generates the following TypeScript definitions:
+
+```typescript
+export class User {
+  free(): void;
+  age: number;
+  name: string;
+}
+```
+
+This ensures that values like `String` are properly cloned before being passed to JavaScript.
+
+### Structs with Functions
+
+You can also return types with methods:
+
+```rust
+#[wasm_bindgen(getter_with_clone)]
+pub struct User {
+    pub name: String,
+    pub age: u32,
+}
+
+#[wasm_bindgen]
+impl User {
+    pub fn say_hello(&self) -> String {
+        format!("Hello, {}!", self.name)
+    }
+}
+```
+
+Which generates the following TypeScript definitions:
+
+```typescript
+export class User {
+  free(): void;
+  say_hello(): string;
+  age: number;
+  name: string;
+}
+```
+
+## How to use `tsify`
+
+While `wasm-bindgen` is ideal for exposing Rust functions and objects to JavaScript, `tsify` is
+better suited for exposing complex data models. The `tsify` approach generates TypeScript bindings
+automatically from Rust structs and works by serializing Rust data into JavaScript-compatible
+objects.
+
+### Basic Types
+
+To return a struct from Rust to JavaScript using `tsify`, derive `Serialize`, `Deserialize`, and
+`Tsify`:
+
+```rust
+use serde::{Deserialize, Serialize};
+#[cfg(feature = "wasm")]
+use {tsify_next::Tsify, wasm_bindgen::prelude::*};
+
+#[derive(Serialize, Deserialize)]
+#[cfg_attr(feature = "wasm", derive(Tsify), tsify(into_wasm_abi, from_wasm_abi))]
+pub struct User {
+    pub name: String,
+    pub age: u32,
+    pub address: Address,
+}
+
+#[derive(Serialize, Deserialize)]
+#[cfg_attr(feature = "wasm", derive(Tsify), tsify(into_wasm_abi, from_wasm_abi))]
+pub struct Address {
+    pub street: String,
+    pub city: String,
+}
+```
+
+Which generates TypeScript interfaces for both `User` and `Address`:
+
+```typescript
+export interface User {
+  name: string;
+  age: number;
+  address: Address;
+}
+
+export interface Address {
+  street: string;
+  city: string;
+}
+
+export function get_user(): User;
```

Comment:
What causes this function to get generated?

Reply:
Woops, no idea how that got there, removing!